django/contrib/sessions/middleware.py

```diff
@@ -47,7 +49,10 @@ def process_response(self, request, response):
                     # Save the session data and refresh the client cookie.
                     # Skip session save for 500 responses, refs #3881.
                     if response.status_code != 500:
-                        request.session.save()
+                        try:
+                            request.session.save()
+                        except UpdateError:
+                            return redirect(request.path)
```

Comment:
Ahh since the user is technically logged out redirecting to the actual request will take to redirect to the login page if desired. That looks correct.